### DISCUSSION TITLE
Question related to operation with multiple items produced

### DISCUSSION THREAD

**Discussion by GOllagnon** [frePPLe/frepple#593] [Q&A]
Dear Forum,

I have a few questions / clarifications points related to situations where one operation produces 4 (more generally many) items based on 2 (more generally a few) consumed items, which usually appears in chemical industry.
(Context : illustration for a chemical customer, in order to make them discover what an APS is, and how/if/what are the benefifs for them.)

Since a moment, I am stuck with a case I am not able to make work, this modelling of operations appears at different steps of their process, so we can summarise that way :
=> there is a distillation operation which produces 4 items from 1 item.
=> then there are independant an dependant demands on these 4 items.
=> the production resource for the distillation is modelled as a time-buckets constrained resource, but the constraint is very low, full-charge only a few days per month.

I built a very small and simple frePPLe model (sales order on 4 end products PF1 - PF4, one component CPT bought from supplier) for that and I noticed that the demand is not totally satisfied

Everything can be summarized on the projected inventory picture :
[3_2_PetitModele_Distillation_InventoryReport.pdf](https://github.com/frePPLe/frepple/files/12000089/3_2_PetitModele_Distillation_InventoryReport.pdf)

I tried different setups with item defined in the operation table to be all of the 4 items, also valued left blank : nothing changed, shortages in all cases.
(Refering to the more complex pilots, it seems to me that the "best" results appear when I setup the "more constrained" item, in order to "pull the demand the more, and get the maximum from the production operation" ... but such a setup is contextual and changes over time ...)
Also I modeled in the same time 4 sisters operations, some the only difference btw them is that one particuliar item PF1 - PF4 as the item delivered by the operation (defined in operation table)

**So my question 1: is frePPLe able to handle that (I think so)?**

I modeled and make simulations using Community version 7.3.0 , but recently I tried the Entreprise version on demo.frepple.com, with some setups cf. inventory policies (service level to very high value 99, huge safety stock), but still shortages
(I tested also the distribution mode for the solver, then the behaviour is OK for that operation, but we can not manage production constraints is such cases, we have the same results as in unconstrained calculation)

**My question 2: it seems to me that in the model I built, the algorithm chooses not to fulfill some orders, is there somewhere a setup to force it to do satisfy all orders (with extra inventory build-up in advance) ?**
**My question 3: does this feature requires inventory optimisation module ?**

(Maybe that I am too closed to the data and the modelling, and that I just need to consider more globally ... it is also the purpose of that message.)

**_Depending of the answer of question 2, I am very interested in contributing to adding that feature, if it is missing ..._**

Please let me know,

Thanks & BR,

Guy.

**Comment by GOllagnon** [comment]
Dear Forum,

I share the small model in excel export.
**My key quesiton remains: why do projected inventories go negative during some moments, whereas they should not, knowing that lots of capacity is available?**

(I reviewed and reviewed again all parameters, everything seems OK from my point of view...)

If someone already faced a smilar situation...

Thanks & BR,

Guy.

[3_1_PetitModele_Distillation.xlsx](https://github.com/frePPLe/frepple/files/12015060/3_1_PetitModele_Distillation.xlsx)

**Comment by GOllagnon** [comment]
Synthesis of the case and of the unexpected behaviour:

A very simple model of operations:
- One component (CPT) is purchased
- One operation of a bucket-day resource with lots of extra capacity consumes CPT and makes 4 items PF1, PF2, PF3, PF4
- Some orders for PF1, PF2, PF3, PF4

When calculating the plan (constrained capacity and lead time), the projected inventories of PF1 and PF2 go negative, whereas the capacity is available

**Comment by hichamlahlou** [comment]
Hi Guy,
There is no reason for the on hand to go negative on a constrained run. We must have some sort of issue at solver level (possibly linked to the fact that you are producing 4 different items). We are looking into it.

**Reply by GOllagnon** [reply]
Hi Hicham,

FYI, it should be useful as an information very high likely related to the same topic, I noticed in some models, when looking in details what are the proposals of frePPLe that :

- e.g. one day, because of an order P24 for item PF1, a requirement is triggered, leading to increasing the inventory of items PF1 and PF2

- a few days later, because of another order P25 for item PF2, another requirements is triggered, which is not relevant, because there were already an increase of stock because of P24; it seems that this increase was not taken into account

Hope it helps,

BR,
Guy.
